tests/functional/nix-copy-ssh-common.sh

```diff
@@ -0,0 +1,72 @@
+source common.sh
+
+clearStore
+clearCache
+
+mkdir -p $TEST_ROOT/stores
+
+# Create path to copy back and forth
+outPath=$(nix-build --no-out-link dependencies.nix)
+
+storeQueryParam="store=${NIX_STORE_DIR}"
+
+realQueryParam () {
+    echo "real=$1$NIX_STORE_DIR"
+}
+
+remoteRoot="$TEST_ROOT/stores/$proto"
+
+clearRemoteStore () {
+    chmod -R u+w "$remoteRoot" || true
+    rm -rf "$remoteRoot"
+}
+
+clearRemoteStore
+
+remoteStore="${proto}://localhost?${storeQueryParam}&remote-store=${remoteRoot}%3f${storeQueryParam}%26$(realQueryParam "$remoteRoot")"
+
+# Copy to store
+
+args=()
+if [[ "$proto" == "ssh-ng" ]]; then
+    # TODO investigate discrepancy
+    args+=(--no-check-sigs)
+fi
+
+[ ! -f ${remoteRoot}${outPath}/foobar ]
+nix copy "${args[@]}" --to "$remoteStore" $outPath
+[ -f ${remoteRoot}${outPath}/foobar ]
+
+# Copy back from store
+
+clearStore
+
+[ ! -f $outPath/foobar ]
+nix copy --no-check-sigs --from "$remoteStore" $outPath
+[ -f $outPath/foobar ]
+
+# Check --substitute-on-destination, avoid corrupted store
+
+if [[ "$proto" == "ssh" ]]; then
+    # FIXME make work for ssh-ng too
```

Comment:
That should work now that https://github.com/NixOS/nix/pull/9600 is merged :confetti_ball: